plugins/backend/catalog-backend-module-aws/src/providers/AWSEBSVolumeProvider.ts

```diff
@@ -38,35 +33,50 @@ const createEbsVolumeConsoleLink = (region: string, volumeId: string): string =>
 /**
  * Provides entities from AWS Elastic Block Store volumes.
  */
-export class AWSEBSVolumeProvider extends AWSEntityProvider {
-  static fromConfig(
-    config: Config,
-    options: {
-      logger: Logger | LoggerService;
-      template?: string;
-      catalogApi?: CatalogApi;
-      providerId?: string;
-      ownerTag?: string;
-      useTemporaryCredentials?: boolean;
-      labelValueMapper?: LabelValueMapper;
-    },
-  ) {
-    const accountId = config.getString('accountId');
-    const roleName = config.getString('roleName');
-    const roleArn = config.getOptionalString('roleArn');
-    const externalId = config.getOptionalString('externalId');
-    const region = config.getString('region');
-
-    return new AWSEBSVolumeProvider(
-      { accountId, roleName, roleArn, externalId, region },
-      options,
-    );
-  }
-
+export class AWSEBSVolumeProvider extends AWSEntityProvider<Volume> {
   getProviderName(): string {
     return `aws-ebs-volume-provider-${this.providerId ?? 0}`;
   }
 
+  protected getDefaultTemplate(): string {
+    return defaultTemplate;
+  }
+
+  protected getResourceAnnotations(
+    resource: Volume,
+    context: { accountId: string; region: string },
+  ): Record<string, string> {
+    const volumeId = resource.VolumeId!;
+    const region = context.region ?? this.region;
+    const consoleLink = createEbsVolumeConsoleLink(region, volumeId);
+    return {
+      [ANNOTATION_VIEW_URL]: consoleLink,
+      [ANNOTATION_EBS_VOLUME_ID]: volumeId,
+    };
+  }
+
+  protected addCustomFilters(env: Environment): void {
+    // Filter: find_name_tag - Extract Name tag from tags array or return fallback
+    env.addFilter(
+      'find_name_tag',
+      (tags: Tag[] | undefined, fallback: string) => {
+        if (!tags || !Array.isArray(tags)) {
+          return fallback;
+        }
+        const nameTag = tags.find(tag => tag.Key === 'Name');
+        return nameTag?.Value || fallback;
+      },
+    );
+
+    // Filter: instance_ids - Extract instance IDs from Attachments array
+    env.addFilter('instance_ids', (attachments: Volume['Attachments']) => {
+      if (!attachments || !Array.isArray(attachments)) {
+        return '';
+      }
+      return attachments.map(a => a.InstanceId).join(', ');
+    });
```

Comment:
![Passing: filters](https://gh-labels.dancrews.com/passing/filters.svg)
Here are some examples of things that felt too opinionated to manipulate ahead of render, and too complex to leave in the template itself. As filters, I could even extend the class and _only_ change the way the filters work, without having to redo _everything_.